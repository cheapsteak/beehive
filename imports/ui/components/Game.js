// Framework
import React from 'react';

// Components
import { Container, Row, Col, Button, CardColumns } from 'reactstrap';
import BeeCard from './BeeCard';

/**
 * The main game element.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const Game = ({ bees = [], removeBee }) => (
  <Container className="game" fluid>
    <Row noGutters>
      <Col>
        <p>Game</p>
      </Col>
    </Row>
    <Row noGutters>
      <Col>
        <p>
          <strong>Total Bees:</strong> {bees.length}
        </p>
      </Col>
    </Row>
    <Row>
      <CardColumns>
        {bees.map(bee => (
          <BeeCard key={bee._id} bee={bee} removeBee={removeBee} />
        ))}
      </CardColumns>
    </Row>
  </Container>
);

export default Game;
