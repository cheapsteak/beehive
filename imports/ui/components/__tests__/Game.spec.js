import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import Game from '../Game';

describe('<Game />', () => {
  let defaultProps;

  describe('Game SnapShot', () => {
    beforeEach(() => {
      defaultProps = {
        bees: [{ _id: 123, name: 'testBee', type: 'worker' }],
        removeBee: jest.fn()
      };
    });

    it('renders game with no bees correctly', () => {
      delete defaultProps.bees;
      const tree = ReactTestRenderer.create(
        <Game {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders game with bees correctly', () => {
      const tree = ReactTestRenderer.create(
        <Game {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
