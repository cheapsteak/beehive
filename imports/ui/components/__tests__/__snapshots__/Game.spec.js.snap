// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> Game SnapShot renders game with bees correctly 1`] = `
<div
  className="game container-fluid"
>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <p>
        Game
      </p>
    </div>
  </div>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <p>
        <strong>
          Total Bees:
        </strong>
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="card-columns"
    >
      <div
        className="bee-card card"
      >
        <img
          alt="worker Bee"
          className="card-img-top"
          src="./images/bees/worker.jpg"
          width="100%"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            testBee
          </h5>
          <h6
            className="card-subtitle"
          >
            worker
          </h6>
          <button
            className="btn btn-danger"
            onClick={[Function]}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Game /> Game SnapShot renders game with no bees correctly 1`] = `
<div
  className="game container-fluid"
>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <p>
        Game
      </p>
    </div>
  </div>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <p>
        <strong>
          Total Bees:
        </strong>
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="card-columns"
    />
  </div>
</div>
`;
